File: apps/worker-notification/logger.ts

export type NotificationStatus = 
"SENT" |
"ALREADY_SENT" |
"NO_OP" |
"DLQ" |
"FAILED" |
"THROTTLED" |
"SKIPPED";

export const logNotification = (input : {
    messageId? : string,
    websiteId : string,
    channel? : string,
    eventType? : string,
    retries? : string,
    status : NotificationStatus,
    reason? : string
})=> {
    console.log(JSON.stringify({
        service : "worker-notification",
        timeStamp : new Date().toISOString(),
        ...input
    }))
}

File: apps/api-go/internal/workers/notification-worker/logger.go
package notificationworker

import (
	"encoding/json"
	"log"
	"time"
)

type NotificationStatus string

const (
	NotifSent        NotificationStatus = "SENT"
	NotifAlreadySent NotificationStatus = "ALREADY SENT"
	NotifNoOp        NotificationStatus = "NO_OP"
	NotifDLQ         NotificationStatus = "DLQ"
	NotifFailed      NotificationStatus = "FAILED"
	NotifThrottled   NotificationStatus = "THROTTLED"
	NotifSkipped     NotificationStatus = "SKIPPED"
)

func LogNotification(
	fields map[string]string,
) {
	logFields := make(map[string]string, len(fields)+2)
	for k, v := range fields {
		logFields[k] = v
	}
	logFields["service"] = "worker-notification"
	logFields["timestamp"] = time.Now().UTC().Format(time.RFC3339)

	b, _ := json.Marshal(logFields)
	log.Println(string(b))
}
